docs/kinesis/kinesis_connector.md

```diff
@@ -0,0 +1,149 @@
+
+## Preparing a Table for Data
+
+To store the data from a stream, we need a table. 
+The columns in the Exasol table have to imitate the data types of the data
+ stored in the stream and also be in the exact order.
+ 
+The table also requires two additional columns to store the Kinesis metadata:  
+    `KINESIS_SHARD_ID VARCHAR(2000), 
+     SHARD_SEQUENCE_NUMBER VARCHAR(2000)`
+These two columns must be at the end of the table.
+ 
+For example, given the following stream data, 
+`{"sensorId": 17,"currentTemperature": 147,"status": "WARN"}`, 
+the Exasol table should be as below:
+
+```sql
+CREATE SCHEMA <schema_name>;
+
+CREATE OR REPLACE TABLE <schema_name>.<table_name> 
+    (SENSORID DECIMAL(18,0), 
+    CURRENTTEMPERATURE DECIMAL(18,0), 
+    STATUS VARCHAR(100), 
+-- Required for importing data from Kinesis
+    KINESIS_SHARD_ID VARCHAR(130), 
+    SHARD_SEQUENCE_NUMBER VARCHAR(2000));
+
+```
+
+## Create ETL UDFs Scripts
+
+Create the following UDF scripts. Please do not change the names of the scripts.
+
+```sql
+--/
+CREATE OR REPLACE JAVA SET SCRIPT KINESIS_METADATA (...) 
+EMITS (KINESIS_SHARD_ID VARCHAR(130), SHARD_SEQUENCE_NUMBER VARCHAR(2000)) AS
+  %scriptclass com.exasol.cloudetl.kinesis.KinesisShardsMetadataReader;
+  %jar /buckets/bfsdefault/kinesis/cloud-storage-etl-udfs-<VERSION>.jar;
+/
+;
+  
+--/
+CREATE OR REPLACE JAVA SET SCRIPT KINESIS_IMPORT (...) EMITS (...) AS
+  %scriptclass com.exasol.cloudetl.kinesis.KinesisShardDataImporter;
+  %jar /buckets/bfsdefault/kinesis/cloud-storage-etl-udfs-<VERSION>.jar;
+/
+;
+    
+--/
+CREATE OR REPLACE JAVA SET SCRIPT KINESIS_PATH (...) EMITS (...) AS
+  %scriptclass com.exasol.cloudetl.kinesis.KinesisImportQueryGenerator;
+  %jar /buckets/bfsdefault/kinesis/cloud-storage-etl-udfs-<VERSION>.jar;
+/
+; 
+```
+
+## Import Data
+
+You can provide AWS credentials using:
+1. UDF parameters 
+1. [Exasol named connection object][exa-connection]. 
+
+We highly **recommend** using connection objects to provide credentials 
+to UDFs so that credentials do not display in audit logs.
+
+### Using Connection Object
+
+First, create a named connection object and encode credentials as key-value
+pairs separated by semicolon (`;`).
+
+`AWS_SESSION_TOKEN` is an optional property. You can remove it from the string
+if you don't need it. 
+
+```sql
+CREATE OR REPLACE CONNECTION KINESIS_CONNECTION
+TO ''
+USER ''
+IDENTIFIED BY 'AWS_ACCESS_KEY=<AWS_ACCESS_KEY>;AWS_SECRET_KEY=<AWS_SECRET_KEY>;AWS_SESSION_TOKEN=<AWS_SESSION_TOKEN>';
+```
+
+Run an import query.
+ 
+```sql
+IMPORT INTO <table_name>
+FROM SCRIPT KINESIS_PATH WITH
+  TABLE_NAME     = '<table_name>'
+  CONNECTION_NAME  = 'KINESIS_CONNECTION'
+  STREAM_NAME    = '<stream_name>'
+  REGION    = '<region>'
+;
+``` 
+
+### Without Connection Object
+
+Provide credentials as properties and run an import query.
+
+**Attention! Providing credentials via propeties is 
+deprecated and will be removed in future releases.**
```

Comment:
Add a deprecation warning to the central README too. Prominently, so that users have no excuse to not read it.

Reply:
done

Reply:
We will do it later